artemis/src/main/java/tech/pegasys/artemis/cli/deposit/EncryptedKeystorePasswordProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.cli.deposit;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import com.google.common.io.Files;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.ParameterException;
+
+public class EncryptedKeystorePasswordProvider {
+  private final String password;
+  private final String environmentVariable;
+  private final File passwordFile;
+  private final String errorPrefix;
+
+  private final CommandSpec spec;
+
+  public EncryptedKeystorePasswordProvider(
+      final CommandSpec spec,
+      final String password,
+      final String environmentVariable,
+      final File passwordFile,
+      final String errorPrefix) {
+    this.spec = spec;
+    this.password = password;
+    this.environmentVariable = environmentVariable;
+    this.passwordFile = passwordFile;
+    this.errorPrefix = errorPrefix;
+  }
+
+  public String retrievePassword() {
```

Comment:
Probably should check that only one of the three options for specifying the password have been provided.  Otherwise it will be very confusing which password actually gets used.